dbt/adapters/spark/connections.py

```diff
@@ -57,15 +70,34 @@ def __post_init__(self):
             )
         self.database = None
 
+        if self.method == SparkConnectionMethod.ODBC and pyodbc is None:
+            raise dbt.exceptions.RuntimeException(
+                f"{self.method} connection method requires "
+                "additional dependencies. \n"
+                "Install the additional required dependencies with "
+                "`pip install dbt-spark[ODBC]`"
```

Comment:
I landed on `pip install dbt-spark[ODBC]` because `pyodbc` is the only python dep that I imagine will require OS dependencies. Also, I think this should like up connection methods (`thrift`, `http`, `odbc`), not connection locations?

Reply:
I buy it!

Reply:
Eventually, I think we might want to try moving `PyHive[hive]` as an extra instead of principal requirement, since it's only necessary for the `http` connection method.